lua/cfc_prop_restoration/core/sv/sv_adinterface.lua

```diff
@@ -1,149 +1,55 @@
-                                    if target.Entity[y] and v.Entity[x].Index == target.Entity[y].Index then
-                                        system[#system + 1] = v
-                                        if #system == 100 then
-                                            fullSystems[#fullSystems + 1] = system
-                                            table.remove( sortingSystems, systemi )
-                                        end
-                                        input[k] = nil
-                                        goto super_loopbreak
-                                    end
-                                end
-                            end
-                        end
-                    end
-                end
-            end
-
-            --Normally skipped by the goto unless no cluster is found. If so, make a new one.
-            local k = next( input )
-            sortingSystems[#sortingSystems + 1] = { input[k] }
-            input[k] = nil
-
-            ::super_loopbreak::
-        end
-    end
-
-    buildSystems( sorted )
-    buildSystems( nocollide )
-
-    local ret = {}
-    for _, system in pairs( fullSystems ) do
-        for _, v in pairs( system ) do
-            ret[#ret + 1] = v
-        end
-    end
+local function canDupe( ply, ent )
+    if not AdvDupe2.duplicator.IsCopyable( ent ) then return false end
+    if areacopy_classblacklist[ent:GetClass()] then return false end
 
-    for _, system in pairs( sortingSystems ) do
-        for _, v in pairs( system ) do
-            ret[#ret + 1] = v
-        end
-    end
-
-    for k, v in pairs( unsorted ) do
-        ret[#ret + 1] = v
-    end
-
-    if #fullSystems ~= 0 then
-        ply:ChatPrint( "DUPLICATOR: WARNING, Number of constraints exceeds 100: (" .. #ret .. "). Constraint sorting might not work as expected." )
-    end
-
-    return ret
+    return ent:CPPIGetOwner() == ply
 end
 
-local function copyPlayerProps( ply )
-    --select all owned props
-    local entities = {}
+function PropRestoration.GetPlayersProps( ply )
+    local props = {}
+
     for _, ent in pairs( ents.GetAll() ) do
-        if playerCanDupeCPPI( ply, ent ) then
-            entities[ent:EntIndex()] = ent
+        if canDupe( ply, ent ) then
+            table.insert( props, ent )
         end
     end
 
-    if table.IsEmpty( entities ) then return end
-
-    local ent = entities[next( entities )]
-    local headEnt = {}
-    headEnt.Index = ent:EntIndex()
-    headEnt.Pos = ent:GetPos()
+    return props
+end
 
-    local entities, constraints = AdvDupe2.duplicator.AreaCopy( entities, headEnt.Pos, true )
+function PropRestoration.Copy( ply )
+    local toCopy = PropRestoration.GetPlayersProps( ply )
 
-    return { entities, constraints, headEnt }
+    local entities, constraints = AdvDupe2.duplicator.AreaCopy( toCopy, copyPos, true )
+    if next( entities ) == nil then return false end
+    return { entities = entities, constraints = constraints }
```

Comment:
Does it not need a headEnt?

Reply:
Head ent is purely for players spawning it in

Reply:
I see these errors in our cl log:
```
[advdupe2] Missing HeadEnt index [119] from Entities table
  1. Decode - addons/advdupe2/lua/advdupe2/sh_codec.lua:515
   2. UploadFile - addons/advdupe2/lua/advdupe2/cl_file.lua:91
    3. DoNodeLeftClick - addons/advdupe2/lua/advdupe2/file_browser.lua:176
     4. unknown - addons/advdupe2/lua/advdupe2/file_browser.lua:1041
```

Reply:
Thats clientside, betterdupe doesnt use headent either and it works fine